backend/api/Services/EchoService.cs

```diff
@@ -88,6 +89,33 @@ public async Task<EchoMission> GetMissionById(int missionId)
             return mission;
         }
 
+        public async Task<EchoMission> GetMissionByPath(string relativePath)
```

Comment:
Given the suggestion that the `Source` should contain the `id` for the echo mission and not the path I think this method might no longer be required.